File: src/transformers/features/http/HttpIsQueryTransformer.ts
import { HttpIsQueryProgrammer } from "../../../programmers/http/HttpIsQueryProgrammer";

import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace HttpIsQueryTransformer {
  export const transform = GenericTransformer.scalar("http.isQuery")(
    (project) => (modulo) => HttpIsQueryProgrammer.write(project)(modulo),
  );
}

File: packages/transform/src/features/http/HttpIsQueryTransformer.go
package http

import (
	corehttp "github.com/samchon/typia/packages/core/src/programmers/http"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var HttpIsQueryTransformer = httpIsQueryTransformerNamespace{}

type httpIsQueryTransformerNamespace struct{}

func (httpIsQueryTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return corehttp.HttpIsQueryProgrammer.Write(call.TypeText)
}
